ct/outline.sh

```diff
@@ -0,0 +1,109 @@
+#!/usr/bin/env bash
+source <(curl -s https://raw.githubusercontent.com/burgerga/ProxmoxVE/add_outline/misc/build.func)
+# Copyright (c) 2024 community-scripts ORG
+# Author: Gerhard Burger (burgerga)
+# License: MIT
+# https://github.com/community-scripts/ProxmoxVE/raw/main/LICENSE
+
+function header_info {
+clear
+cat <<"EOF"
+   ____        __  ___
+  / __ \__  __/ /_/ (_)___  ___
+ / / / / / / / __/ / / __ \/ _ \
+/ /_/ / /_/ / /_/ / / / / /  __/
+\____/\__,_/\__/_/_/_/ /_/\___/
+
+EOF
+}
+header_info
+echo -e "Loading..."
+APP="Outline"
+var_disk="8"
+var_cpu="2"
+var_ram="4096"
+var_os="debian"
+var_version="12"
+variables
+color
+catch_errors
+
+function default_settings() {
+  CT_TYPE="1"
+  PW=""
+  CT_ID=$NEXTID
+  HN=$NSAPP
+  DISK_SIZE="$var_disk"
+  CORE_COUNT="$var_cpu"
+  RAM_SIZE="$var_ram"
+  BRG="vmbr0"
+  NET="dhcp"
+  GATE=""
+  APT_CACHER=""
+  APT_CACHER_IP=""
+  DISABLEIP6="no"
+  MTU=""
+  SD=""
+  NS=""
+  MAC=""
+  VLAN=""
+  SSH="no"
+  VERB="no"
+  echo_default
+}
+
+function update_script() {
+header_info
+check_container_storage
+check_container_resources
+if [[ ! -d /opt/outline ]]; then msg_error "No ${APP} Installation Found!"; exit; fi
+RELEASE=$(curl -s https://api.github.com/repos/outline/outline/releases/latest | grep "tag_name" | awk '{print substr($2, 2, length($2)-3) }')
+if [[ "${RELEASE}" != "$(cat /opt/${APP}_version.txt)" ]] || [[ ! -f /opt/${APP}_version.txt ]]; then
+  msg_info "Stopping ${APP}"
+  systemctl stop outline
+  msg_ok "Stopped ${APP}"
+
+  msg_info "Updating ${APP} to ${RELEASE} (Patience)"
+  cd /opt
+  cp /opt/outline/.env /opt/.env
+  mv /opt/outline /opt/outline_bak
```

Comment:
True, that is in case the update process fails halfway (like happened to me using the linkwarden update). I adapted it from https://github.com/community-scripts/ProxmoxVE/blob/fab35a61902384a1425f6211a6511f489cc42d69/ct/linkwarden.sh#L88 (but here the `linkwarden_bak` is never created, so that won't work for sure :P)

Reply:
Is there any danger for data loss when removing outline source folder though? Doesn't it use postgres database and redis?
(feel free to remove that unused linkwarden line)

Reply:
No risk of data loss indeed, but if the update fails because of upstream changes, it would be good to be able to go back to the previous version without rebuilding (because there is then no way to build it using the community script, since it defaults to the latest release, right?)